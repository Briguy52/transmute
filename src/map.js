import curry from "./curry";

/**
 * Create a new Iterable by applying `mapper` to each item in `subject`.
 *
 * @example
 * // returns List [ 2, 3, 4 ]
 * map(
 *   (val) => val + 1,
 *   List.of(1, 2, 3)
 * );
 *
 * @param  {Function} mapper applied to each item in `subject`.
 * @param  {Iterable} subject the Iterable to map.
 * @return {Iterable} with each value of `subject` updated with mapper.
 */
function map(mapper, subject) {
  return subject.map(mapper);
}

export default curry(map);
