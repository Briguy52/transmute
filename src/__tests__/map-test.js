import { List, Map } from "immutable";
import map from "../map";

describe("transmute/map", () => {
  const mapper = map(n => n + 1);

  it("maps a List", () => {
    expect(mapper(List.of(1, 2, 3))).toMatchSnapshot();
  });

  it("maps a Map", () => {
    expect(mapper(Map.of("one", 1, "two", 2, "three", 3))).toMatchSnapshot();
  });
});
