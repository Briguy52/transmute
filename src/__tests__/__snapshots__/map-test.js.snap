// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/map maps a List 1`] = `
Immutable.List [
  2,
  3,
  4,
]
`;

exports[`transmute/map maps a Map 1`] = `
Immutable.Map {
  one: 2,
  two: 3,
  three: 4,
}
`;
